crates/viewer/re_space_view_spatial/src/visualizers/boxes3d.rs

```diff
@@ -78,32 +104,67 @@ impl Boxes3DVisualizer {
 
             let centers = clamped_or(data.centers, &Position3D::ZERO);
             let rotations = clamped_or(data.rotations, &Rotation3D::IDENTITY);
-            for (i, (half_size, &center, rotation, radius, &color)) in
-                itertools::izip!(data.half_sizes, centers, rotations, radii, &colors).enumerate()
+            for (
+                instance_index,
+                (half_size, &center, rotation, radius, &surface_color, &line_color),
+            ) in itertools::izip!(
+                data.half_sizes,
+                centers,
+                rotations,
+                radii,
+                &surface_colors,
+                &line_colors
+            )
+            .enumerate()
             {
+                let instance = Instance::from(instance_index as u64);
+                // Transform from a centered unit cube to this box.
+                let transform = glam::Affine3A::from_scale_rotation_translation(
+                    glam::Vec3::from(*half_size) * 2.0,
+                    rotation.0.into(),
+                    center.into(),
+                );
+
                 obj_space_bounding_box.extend(half_size.box_min(center));
                 obj_space_bounding_box.extend(half_size.box_max(center));
 
-                let center = center.into();
+                if line_color != Color32::TRANSPARENT {
+                    let box3d = line_batch
+                        .add_box_outline_from_transform(transform)
+                        .color(line_color)
+                        .radius(radius)
+                        .picking_instance_id(PickingLayerInstanceId(instance_index as _));
+
+                    if let Some(outline_mask_ids) = ent_context.highlight.instances.get(&instance) {
+                        box3d.outline_mask_ids(*outline_mask_ids);
+                    }
+                }
 
-                let box3d = line_batch
-                    .add_box_outline_from_transform(
-                        glam::Affine3A::from_scale_rotation_translation(
-                            glam::Vec3::from(*half_size) * 2.0,
-                            rotation.0.into(),
-                            center,
+                if surface_color != Color32::TRANSPARENT {
+                    let proc_mesh_key = proc_mesh::ProcMeshKey::Cube;
+                    let Some(solid_mesh) = ctx
+                        .viewer_ctx
+                        .cache
+                        .entry(|c: &mut proc_mesh::SolidCache| c.entry(proc_mesh_key, render_ctx))
+                    else {
+                        return Err(SpaceViewSystemExecutionError::DrawDataCreationError(
+                            "Failed to allocate solid mesh".into(),
+                        ));
+                    };
+
+                    obj_space_bounding_box =
+                        obj_space_bounding_box.union(solid_mesh.bbox.transform_affine3(&transform));
```

Comment:
that seems both wrong and unnecessary: we already extended the _object space_ bounding box with the half extents further up. This actually extends it with a world space bounding box if I'm not mistaken (which we then later gets converted/expanded to world space again and added to the list of world space bounding boxes).

The only reason I can think of for this is to account for the mesh being slightly larger than the "analytical" bounding box. But I would have assume if anything it would be smaller?
In that case is should be just `obj_space_bounding_box.union(solid_mesh.bbox)`

Reply:
I agree there's a mistake here (this was a cut-and-paste-o from `Ellipsoids` where it is more important), but note the *existing* code is not correct — `obj_space_bounding_box.extend(half_size.box_min(center));` does not account for the `rotations` from the archetype. Filed #6973 for that.

Reply:
> This actually extends it with a world space bounding box if I'm not mistaken

No, I think this is correct. The variable named `transform` here is the per-*instance* transform defined by `centers`, `half_size`, and `rotations`, per-box/ellipsoid — not the `world_from_entity` transform. We need to apply *both* transforms eventually.

Reply:
Pushed new changes. I'm still doing this, and I've checked that it produces the right result at least for `box3d_batch`.

Reply:
ah nice. Got confused there! And thanks for filing that issue